src/extension.ts

```diff
@@ -565,6 +565,47 @@ export async function activate(context: vscode.ExtensionContext) {
     });
   });
 
+  registerIDFCommand("espIdf.searchInEspIdfDocs", async () => {
+    const docsVersions = [
+      "v4.0.1",
+      "v4.0",
+      "v3.3.2",
+      "v3.3.1",
+      "v3.3",
+      "v3.2.3",
+      "v3.2.2",
+      "v3.1.7",
+      "v3.1.6",
+      "v3.1.5",
+      "v3.0.9",
+    ];
+    const idfPath =
+      idfConf.readParameter("idf.espIdfPath") || process.env.IDF_PATH;
+    let idfVersion = "v" + (await utils.getEspIdfVersion(idfPath));
+    if (docsVersions.indexOf(idfVersion) === -1) {
+      const idfTarget =
+        idfConf.readParameter("idf.adapterTargetName") || "esp32";
+      idfVersion = `latest/${idfTarget}`;
+    }
+    const currentEditor = vscode.window.activeTextEditor;
+    if (currentEditor) {
+      let selection = currentEditor.document.getText(currentEditor.selection);
+      if (!selection) {
+        const range = currentEditor.document.getWordRangeAtPosition(
+          currentEditor.selection.active
+        );
+        selection = currentEditor.document.getText(range);
```

Comment:
one potential problem is that .... user can select any text and it will open the docs 

One thing would be better @igrr if we can have a param passed to the doc server like `https://docs.espressif.com/projects/esp-idf/en/${idfVersion}/search.html?q={query}&from=vscode.esp-idf-extension`  ... so that doc could render a page if no ref is found that from vscode you can only query valid api or stuff ...

In advance case we can parse the AST and find refs and then only query certain tokens but may be it won't solve the problem still

Reply:
Any idea about this ?

Reply:
One possible thing to do is write a parser to parse and package document nodes into a json file which can then be use to lookup the token and if found will then show the content menu `espIdf.searchInEspIdfDocs` and then open the link. But this can be done in later iterations IMO.

```js
selection = currentEditor.document.getText(range);
  if (await util.existInDoc(selection)) { ... }
}
```
JSON file could be simply an array
```json
tokens = [ ... ]
```

Reply:
The problem is that JSON file would be huge, it is basically crawling and indexing the whole esp-idf documentation...

Reply:
True, could leave it be for now, in future releases we can have some kind of optimize/compressed list of valid keyword and search mechanism